pxtlib/tilemap.ts

```diff
@@ -1041,17 +1186,25 @@ namespace pxt {
         }
     }
 
+    export function validateAssetName(name: string) {
+        if (!name) return false;
+
+        // Covers all punctuation/whitespace except for "-", "_", and " "
+        const bannedRegex = /[\u0000-\u001f\u0021-\u002c\u002e\u002f\u003a-\u0040\u005b-\u005e\u0060\u007b-\u007f]/
```

Comment:
I could maybe see `.` being used to 'namespace' assets, and maybe `?!` for kids who are excited about their sprites? But fine either way